docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+
+Step 3. Next, execute layer will call `action.act()` which causes StatsAction to begin its `act()` operation. If `isMod` flag is set, `act()` will search for the user specified module and get the list of tasks tagged to it.
+
+Step 5. Once the list of task is obtain, the operation will loop through the task list and count the number of completed task followed by generating a ratio. 
+
+Step 6. This ratio will be passed into `roundedRatioBar` to return *String* of a rounded ratio to 1 decimal place enclosing it in square brackets. 
+
+Step 7. Now `StatsAction` is completed and it will return this string back to `Execute` for to be printed through `UI`. 
+
+
+**Design consideration:**
+
+**Aspect : How statistics executes**
+ - **Alternative 1 (current choice):** Create a separate class and get list of tasks/taken modules' task and scan through them to calculate statistics
+	 - Pros: Reduces Coupling and increase testability as a software unit itself. 
+	 - Cons: May have performance issues in terms of memory usage 
+
+ - **Alternative 2:** initialize statistics as zero and each task contains an aspect called statistics
+	 - Pros: Will use less memory since the task itself will be deleted. 
+	 - Cons: Stats will be updated constantly even though we do not need it. 
+
+### 4.3 Checker Feature 
+
+The checker mechanism is facilitated by the utility class `Checker`. It is an independent class on its own without extensions and is stored under the `Data` package of our app. The class implements the following operations: 
+
+ - `checkDuplicates()`- Calls the checkClash method and return the status of boolean variable `isClash` .
+ - `checkClash(ArrayList< item >, Item)`- Updates `isClash` once a duplicate item is found in the list.
+ - `checkRecurrenceDate(Task)` - Checks if the current date is beyond the stated date in the list, and provides a new update for the date recurring date.
+
+Given below is an example usage scenario and how the checker mechanism behaves at each step. 
+
+![here](Images/Checker_Diagram.png)
+
+Step 1. A new `Deadline` object is created and needs to be added to the existing list of task. Hence it calls `addTask()` method under `Data` class. 
+
+Step 2. Data instantiates a Checker with its existing list by calling its constructor, and the task to be added to the list 
+
+Step 3. Data proceeds to call the `checkRecurrenceDate(Task)` of the Checker class, to get a newDate if today's date is beyond the stated weekly date.
+ 
+Step 4. If `newDate` is not `null`it shows that there is a new updated date. Therefore, we proceed to update the object with our updated weekly date. 
+
+Step 5. Now we proceed to call `checkDuplicates()` of Checker class. 
+
+Step 6. If `false` , there is no duplicates in the existing list, and the task can be safely added. Otherwise, no action will be taken. 
+
+**Design consideration: 
+Aspect: How checker executes**
```

Comment:
This part looks problematic on the page. I guess you forgot to add "***" at the right places?
![image](https://user-images.githubusercontent.com/59434361/97668780-af78c200-1abd-11eb-928e-253d11cc1d82.png)